.github/workflows/backup-and-mirror-charts.yaml

```diff
@@ -0,0 +1,42 @@
+name: "Backup and then Mirror Helm Charts"
+
+on:
+  # push: # runs on any branch push
+  # schedule:
+  #   - cron: "0 0 * * 1" # runs every Monday at midnight
+  workflow_dispatch: # runs whenever the workflow is dispatched via the UI
+
+env:
+  REGISTRY_SOURCE: "https://charts.konstruct.io/"
+  REGISTRY_DESTINATION: "https://charts.konstruct.io/"
+
+permissions:
+  contents: write
+
+jobs:
+    # env context is not accessable in called workflow
+  # https://github.com/orgs/community/discussions/26671
+  # https://stackoverflow.com/questions/73305126/passing-env-variable-inputs-to-a-reusable-workflow
+  pass-env-var-to-workflow:
+    runs-on: ubuntu-latest
+    outputs:
+      registry-source: ${{ steps.step1.outputs.REGISTRY_SOURCE }}
+      registry-destination: ${{ steps.step2.outputs.REGISTRY_DESTINATION }}
+    steps:
+      - id: step1
+        run: echo "REGISTRY_SOURCE=${REGISTRY_SOURCE}" >> "$GITHUB_OUTPUT"
+      - id: step2 
+        run: echo "REGISTRY_DESTINATION=${REGISTRY_DESTINATION}" >> "$GITHUB_OUTPUT"
```

Comment:
Can I propose to you a different recommendation? The goal of the env var was just to be able to reference a version on a single place.

If we create sub-workflows that can be embedded into another, we can make the value to be specified directly when executing it, so instead of all of this above, just leave the bottom part as:

```yaml
  download-charts:
    uses: konstructio/charts-mirror/.github/workflows/tpl-download-charts.yaml@mirror-only
    needs: pass-env-var-to-workflow
    with:
      registry-source: "https://charts.konstruct.io"

  mirror-charts:
    uses: konstructio/charts-mirror/.github/workflows/tpl-mirror-charts.yaml@mirror-only
    needs: pass-env-var-to-workflow
    with:
      registry-destination: "https://charts.konstruct.io"
```

It's a very simple solution, achieves the same goal (the value is set once, maybe twice), and it's often how we leverage other versions elsewhere: most GitHub actions with an app in them support specifying a version as part of the `with:` options.

That would also tremendously improve the readability. While this works, I can't but think there's a bit of black magic involved that someone with some deep knowledge would have to leverage. I was hoping for this repo to be an example on how to use our forked `helm-mirror` so keeping it simple might show potential users how "easy" it is to achieve what we want.

(And as a total side note, maybe a better goal could simply be creating a GitHub Action for `helm-mirror` 😉)

What do you think?

Reply:
I agree boss. It is a convoluted way to pass variable downstream and it's definitely not readable.

Let's discuss more about helm-mirror GitHub action in a separate issue.